File: src/Web/OnlineGameConverter/onlinegameconverter.client/Library/Measurements/DataConsumer.ts
import { CategoryObject } from "../CategoryObject";
import type { ICategoryObject } from "../Interfaces/ICategoryObject";
import type { IDesktop } from "../Interfaces/IDesktop";
import type { IPostSetArrow } from "../Interfaces/IPostSetArrow";
import type { IPrintedObject } from "../Interfaces/IPrintedObject";
import type { IPrinter } from "../Interfaces/IPrinter";
import type { IDataConsumer } from "./Interfaces/IDataConsumer";
import type { IMeasurements } from "./Interfaces/IMeasurements";
import type { ITimeMeasurementConsumer } from "./Interfaces/ITimeMeasurementConsumer";
import type { ITimeMeasurementProvider } from "./Interfaces/ITimeMeasurementProvider";

export class DataConsumer extends CategoryObject implements IDataConsumer, IPostSetArrow, ITimeMeasurementConsumer, IPrintedObject
{
    constructor(desktop: IDesktop, name: string)
    {
        super(desktop, name);
        this.typeName = "DataConsumer";
        this.types.push("DataConsumer");
        this.types.push("IDataConsumer");
        this.types.push("IPostSetArrow");
        this.types.push("ITimeMeasurementConsumer");
        this.types.push("IPrintedObject");
        this.tms = this;
        this.dataConsumer = this;
    }

    print(printer: IPrinter): void {
        for (var m of this.measurements) {
            let co = m as unknown as ICategoryObject;
            let s = co.getCategoryObjectName() + "\t";
            let n = m.getMeasurementsCount();
            for (let i = 0; i < n; i++) {
                var mm = m.getMeasurement(i);
                var v = mm.getMeasurementValue();
                s += v + "\t";
            }
            printer.print(s);
        }
    }

    getInternalTime(): number
    {
        var tm = this.timeMeasurement;

        return tm.getTime();
    }

  
    tms!: ITimeMeasurementConsumer; 

    timeMeasurement !: ITimeMeasurementProvider;

    success: boolean = true;

    protected dataConsumer !: IDataConsumer;

   
    
    getTimeMeasurement(): ITimeMeasurementProvider {
        return this.timeMeasurement;
    }

    setTimeMeasurement(measurement: ITimeMeasurementProvider): void {
        this.timeMeasurement = measurement;            ;
    }

    postSetArrow(): void {

    }

    private measurements: IMeasurements[] = [];

   
    getAllMeasurements(): IMeasurements[] {
        return this.measurements;
    }
    addMeasurements(item: IMeasurements): void {
        this.measurements.push(item);
    }

}

File: Java_Generated/Java/Trading1/app/src/main/java/measurements/DataConsumer.java
package measurements;

import category_theory.CategoryObject;
import category_theory.interfaces.ICategoryObject;
import diagram.interfaces.IDesktop;
import error_handler.interfaces.ICheck;
import error_handler.interfaces.ICheckHolder;
import general_service.Entry;
import general_service.Performer;
import general_service.interfaces.*;
import measurements.interfaces.IDataConsumer;
import measurements.interfaces.IMeasurements;
import measurements.time.interfaces.ITimeMeasurementConsumer;
import measurements.time.interfaces.ITimeMeasurementProvider;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class DataConsumer extends CategoryObject implements IDataConsumer, IPostSetArrow,
        ITimeMeasurementConsumer, IPrintedObject, ICheckHolder {

    protected Performer performer = new Performer();
    protected measurements.Performer mPerformer = new measurements.Performer();

    ITimeMeasurementConsumer tms;

    ITimeMeasurementProvider timeMeasurement;

    protected boolean success = true;

    protected IDataConsumer dataConsumer;

    protected IValueSetterFactory valueSetterFactory;

    protected List<IMeasurements> measurements = new ArrayList<>();

    protected List<IMeasurements> dependent = new ArrayList<>();

    protected List<Object> list = new ArrayList<>();

    protected Object variable;

    public DataConsumer(String name, IDesktop desktop) {
        super(name, desktop);
        valueSetterFactory = desktop.getValueSetterFactory();
    }

    protected void init() {

    }


    @Override
    public IMeasurements[] getAllMeasurements() {
        var size = measurements.size();
        IMeasurements[] m = new IMeasurements[size];
        measurements.toArray(m);
        return m;
    }

    @Override
    public void addMeasurements(IMeasurements item)
    {
        measurements.add(item);
        mPefrformer.GetDependent(measurements, list, dependent);
    }

    @Override
    public void postSetArrow() {

    }

    @Override
    public ITimeMeasurementProvider getTimeMeasutement() {
        return timeMeasurement;
    }

    @Override
    public void setTimeMeasurement(ITimeMeasurementProvider provider) {
        timeMeasurement = provider;
    }

    @Override
    public double getInternalTime() {
        return 0;
    }

    protected String[] copyMap(Map<String, Entry<Object, Object>> input, Map<String, IValueSetter> output) {
        return mPefrformer.copyMap(input, output, valueSetterFactory);
    }


    protected measurements.Performer mPefrformer = new measurements.Performer();

    @Override
    public void print(IPrinter printer) {
        var m = getAllMeasurements();
        for (var mea : m) {
            ICategoryObject co = (ICategoryObject) mea;
            var s = co.getCategoryObjectName() + " ";
            int n = mea.getMeasurementsCount();
            for (int i = 0; i < n; i++) {
                s += mea.getMeasurement(i) + " ";
            }
            printer.print(s + "\n");
        }
    }

    @Override
    public ICheck getChecker() {
        return checker;
    }

    @Override
    public void setChecker(ICheck check) {
        checker = check;

    }
}